docs/src/examples/components/Form/Types/FormExample.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { Form, Input, Button } from '@stardust-ui/react'
+
+const FormExample = () => (
+  <Form
+    onSubmit={() => {
+      alert('Form submitted')
+    }}
+  >
+    <Form.Field
+      label="First name"
+      controlType={Input}
```

Comment:
this seems to have `as` semantics - maybe we should name this prop like `controlAs`?

Reply:
also, thinking that for the first stages maybe it would be fine to eliminate this prop altogether - and just use `control` slot for that:

```jsx
 <Form.Field
      label="Last name"
      name="lastName"
      id="last-name"
      required={true}
      control={{ as: 'input' }}
```

A benefit of this approach would be that we will eliminate any possibilities for collisions of different prop values come from different places. What do you think?

Reply:
I added this as a specific property because we need it for actually creating the slot used here:
```jsx
    const factoryMethod = createSlotFactory(controlType, name => ({ name }))
    const controlElement = factoryMethod(control || {}, {
      defaultProps: { required, id, name, type, styles: styles.control },
      render: renderControl,
    })
```
We are using the Slot as this can be some HTML element, or component. If we use this form the 'as' property, then it is becoming kind a tricky because this can be object, component, some primitive type etc. What do you think? I am okay with renaming it, but yeah, let's decide whether it should exists as a prop first.

Reply:
We might right away use the Slot.create and propagate the control shorthand there, but this was the proposed way for using the Slot create method. I am open for other suggestions.